### PR TITLE
Divide the apps modules into direct CLI command modules and support library modules

### PR DESCRIPTION
**Pull request by levitte**
For starters, everything in apps includes apps.h, because that one declares apps internal library routines.  However, progs.h doesn't declare library routines, but rather the main commands and their options, and there's no reason why the library modules should include it.

So, remove the inclusion of progs.h from apps.h and add that inclusion in all command source files.

Then, and mostly for practical purposes, we place the support library modules in a
private, static library.

Finally, this greatly simplifies our dealing with apps stuff in the test C framework

### CODE REVIEW

File: apps/openssl.c [hunk 1]
```diff
@@ -29,6 +29,7 @@
 #endif
 #define INCLUDE_FUNCTION_TABLE
```
Comment (richsalz) [inline apps/openssl.c]: This only affects progs.h so while you're here, move the define.

**Review by richsalz — APPROVED** [review]
one nit, approve assuming you'll fix it.

File: apps/openssl.c (hunk 1 above)
Comment (levitte) [reply apps/openssl.c]: I'm not sure I understand where you think it should be moved...

Comment (levitte) [reply apps/openssl.c]: Oh wait, just down one line?  Yeah I can do that

**Comment by levitte** [comment]
Ok, `INCLUDE_FUNCTION_TABLE` definition moved one line... and that was pre-approved, cool :-)

**Comment by levitte** [comment]
So, errrr, Appveyor seems to dislike *something*...  I need to dig a bit, please expect some extra changes, hopefully small

**Comment by levitte** [comment]
Ah, I see why, there's platform specific init code that doesn't get linked into the app when it's in the library.  *sigh* time for a rethink then.

**Comment by levitte** [comment]
Ok, I think I've found a solution...

**Comment by richsalz** [comment]
reconfirmed once CI's build.

**Comment by levitte** [comment]
Noted. Thanks :-)

**Comment by levitte** [comment]
Ok, CIs finally happy.

**Comment by levitte** [comment]
Merged. dab2cd68e7cc304c9b1a4e7cee18a98711771a53 to 0ac9e9ff5de278fed705a2bf090ed92b5911656b

**Pull request closed without merging**